src/Microsoft.TestPlatform.CrossPlatEngine/Client/Parallel/ParallelProxyDiscoveryManager.cs

```diff
@@ -248,5 +276,19 @@ private void DiscoverTestsOnConcurrentManager(IProxyDiscoveryManager proxyDiscov
                 EqtTrace.Verbose("ProxyParallelDiscoveryManager: No sources available for discovery.");
             }
         }
+
+        /// <summary>
+        /// Gets all sources which were not discovered(skipped), because of cancelation/abortion of discovery
+        /// </summary>
+        private void getAllNotDiscoveredSources()
+        {
+            lock (enumeratorLockObject)
+            {
+                while (this.sourceEnumerator != null && this.sourceEnumerator.MoveNext())
```

Comment:
Why do we MoveNext before we get to process the current event ? Doesn't that mean we'll skip the current element from processing ?

Reply:
Based on this [docs](https://docs.microsoft.com/en-us/dotnet/api/system.collections.ienumerator.movenext?view=net-5.0#remarks) initial pointer points before first element, so we should call `MoveNext` to get first element.